File: src/main/java/net/vortetty/pulaskisandshaxes/config/modmenuIntegration.java
//
// A layer between kotlin and modmenu to make everything work, errors if this isn't here
//
package net.vortetty.pulaskisandshaxes.config;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import me.shedaniel.clothconfig2.api.ConfigBuilder;
import net.minecraft.client.gui.screen.Screen;
import net.minecraft.text.TranslatableText;
import net.vortetty.pulaskisandshaxes.pulaskisandshaxes;

public class modmenuIntegration implements ModMenuApi {
    Screen getConfigScreen(Screen parent) {
        ConfigBuilder builder = ConfigBuilder.create()
                .setParentScreen(parent)
                .setTitle(new TranslatableText("title.pulaskisandshaxes.config"));

        pulaskisandshaxes.Companion.getConfig().generateConfigScreen(builder);

        return builder.build();
    }

    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return parent -> getConfigScreen(parent);
    }
}

File: src/main/kotlin/net/vortetty/pulaskisandshaxes/config/modMenuIntegration.kt
package net.vortetty.pulaskisandshaxes.config

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import me.shedaniel.autoconfig.AutoConfig
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.client.gui.screen.Screen

@Environment(EnvType.CLIENT)
class modMenuIntegration : ModMenuApi {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        return ConfigScreenFactory { parent -> AutoConfig.getConfigScreen(configuration::class.java, parent).get() as Screen }
    }
}